static/elements/chromedash-upcoming.js

```diff
@@ -52,12 +52,110 @@ class ChromedashUpcoming extends LitElement {
       loginUrl: {type: String},
       starredFeatures: {type: Object}, // will contain a set of starred features
       cardWidth: {type: Number},
+      lastFetchedOn: {type: Number},
+      lastMilestoneVisible: {type: Number},
+      milestoneArray: {type: Array},
+      milestoneInfo: {type: Object},
+      cardsToFetchInAdvance: {type: Number},
     };
   }
 
+  set lastFetchedOn(val) {
+    let oldVal = this._lastFetchedOn;
+    this._lastFetchedOn = val;
+    this.fetchNextBatch().then(()=>{
+      this.requestUpdate('lastFetchedOn', oldVal);
+    }).catch(() => {
+      alert('Some error occurred. Please refresh the page or try again later.');
+    });
+  }
+
+  get lastFetchedOn() {
+    return this._lastFetchedOn;
+  }
+
   constructor() {
     super();
     this.cardWidth = this.computeWidthofCard();
+    this.starredFeatures = new Set();
+    this.milestoneArray = [];
+    this.milestoneInfo = {};
+    this.cardsToFetchInAdvance = 3;
+  }
+
+  async fetchNextBatch() {
+    // promise to fetch next milestones
+    const nextMilestones = window.csClient.getSpecifiedChannels(this._lastFetchedOn+1+1,
+      this._lastFetchedOn+1+this.cardsToFetchInAdvance);
+
+    let tempMilestoneArray = [];
+    for (let i = 1; i <= this.cardsToFetchInAdvance; i++) {
+      tempMilestoneArray.push(this._lastFetchedOn+1+i);
+    }
+
+    // promise to fetch features in next milestones
+    let milestoneFeaturePromise = {};
+    tempMilestoneArray.forEach((milestone) => {
+      milestoneFeaturePromise[milestone] = window.csClient.getFeaturesInMilestone(milestone);
+    });
+
+    let tempMilestoneInfo;
+    let milestoneFeatures = {};
+
+    try {
+      tempMilestoneInfo = await nextMilestones;
+
+      for (let milestone of tempMilestoneArray) {
+        milestoneFeatures[milestone] = await milestoneFeaturePromise[milestone];
+      }
+    } catch (err) {
+      throw (new Error('Unable to load Features'));
+    }
+
+
```

Comment:
Style: Only use a single blank line in the body of a method or function.  Use double blank lines between top-level declarations.

Reply:
Thanking you for pointing this out. 
Done. See [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1458/commits/f4600559c8861551d9edaf934cfe2d075b855f7a) commit